cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -160,6 +160,9 @@ func Run(s *options.CMServer) error {
 	if err != nil {
 		return err
 	}
+	// it is more safe to use hostname plus pod name
+	// see https://github.com/kubernetes/kubernetes/issues/22775
+	id = id + "-" + os.Getenv("POD_NAME")
```

Comment:
You can declare a containerPort. If the pod is running as hostNetwork, this will have the same effect as a hostPort conflict so the pods will never be scheduled together. I think that also solves 2. Does that work?

I'm also not opposed to being able to override holder identity. I just think it should be "hostname OR something" not "hostname AND something".